### ISSUE TITLE
Yarn cache not found

### ISSUE THREAD

**Issue by mdgeorge4153** [actions/setup-node#441]
**Description:**
I get repeated misses for the yarn cache: ```Run actions/setup-node@v2
  with:
    node-version: 16.x
    check-latest: true
    cache: yarn
    always-auth: false
    token: ***
Attempt to resolve the latest version from manifest...
Resolved as '16.14.0'
Found in cache @ /opt/hostedtoolcache/node/16.14.0/x64
/usr/local/bin/yarn --version
1.22.17
/usr/local/bin/yarn cache dir
/home/runner/.cache/yarn/v6

yarn cache is not found```

**Action version:**
v2

**Platform:**
- [X] Ubuntu
- [ ] macOS
- [ ] Windows

**Runner type:**
- [X] Hosted
- [ ] Self-hosted

**Tools version:**
node-version: 16.x

**Repro steps:**  
Public repo: https://github.com/Certora/rocket-joe-1/commit/ab70fe2af9afd51447945e0a3fcf6ad7745eff98

**Expected behavior:**
Cache hit

**Actual behavior:**
Cache miss after repeated reruns

**Comment by mdgeorge4153** [comment]
This seems to be similar to bug #410 , but I have a public repo.

**Comment by mdgeorge4153** [comment]
This also suddenly resolved for me.  I think what may have been happening is that the cache gets updated later in the job, and if another job fails it fails fast and doesn't get around to doing the cleanup.  I notice that the same issue was happening with setup-python and the pip cache, and resolved itself at the same time.

**Comment by mdgeorge4153** [comment]
Actually it has reoccurred; I guess it is nondeterministic

**Comment by dmitry-shibanov** [comment]
Hello @mdgeorge4153. Thank you for your report. Does the error still reproduce?

**Comment by marko-zivic-93** [comment]
I will close this issue. 
If you experience this issue again, please feel free to leave a comment here or to create a new issue.

**Comment by just-Bri** [comment]
Preface: I'm on yarn@1.22.19 - can't upgrade to 3 because of snyk :/

I've been seeing this issue since enabling `cache: 'yarn'`
Private repo unfortunately.  

I've tried doing a couple small pushes with no package.json or yarn.lock changes to see if those changes were causing the cache not to be found but no luck so far.

**Comment by damhonglinh** [comment]
I had the same issue and found a way to solve this. 

I checked the log in the later job `Post Run actions/setup-node@v3` and found that there was a warning `/bin/tar: unrecognized option: posix` so the cache couldn't be built. It was because I was using `alpine` Linux image ([more explanation here](https://stackoverflow.com/a/64187955/1815779)). 

So changing to use normal Linux image solved this for me.

**Comment by wildstyles** [comment]
Seems it caches only after job is succeeded once
<img width="1280" alt="image" src="https://github.com/actions/setup-node/assets/28781969/22801eff-cb17-4ef2-a269-abe6fd0f0deb">
